manifest: Update Matter SDK revision to pull RAM optimizations (#12313)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

> * Enabled :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option by default, including for builds without factory data support.

Reviewer: This suggestion would replace Lines 106 to 116:
```
* Updated:

  * Matter over Thread samples so that the OpenThread shell is disabled by default.
  * The :kconfig:option:`CONFIG_CHIP_FACTORY_RESET_ERASE_NVS` Kconfig option to be enabled by default, including for builds without factory data support.
    The firmware now erases all flash pages in the non-volatile storage during a factory reset, instead of just clearing Matter-related settings.
  * The RAM usage based on test measurements.
    After the following optimizations, the RAM usage decreased by around 12-20% on all supported boards:

    * Reduced the number of network and Matter stack buffers and packets.
    * Disabled SSL server support.
    * Reduced the Main, Matter and OpenThread stack sizes.
    * Reduced the Mbed TLS heap size.
    * Improved the buffer usage of the nRF700X driver for Matter.
    * Reduced the size of the Matter event queue.
```
Tried my best to take into account comments by @Damian-Nordic

Author: Thank you!

  The firmware now erases all flash pages in the non-volatile storage during a factory reset, instead of just clearing Matter-related settings.

* RAM usage optimizations:

  * Reduced number of network and Matter stack buffers and packets.
  * Disabled SSL server support.

>   * Reduced the Main, Chip, mbedTLS, and OpenThread stack sizes according to the measurements.

Reviewer 2: ```suggestion
  * Reduced the stack sizes for Main, Matter and OpenThread threads.
```
"according to the measurements" applies to all optimizations, so maybe this should be added before or after the bullet list?
Also, for mbedTLS we reduced the heap, not stack.

  * Optimized buffer usage of nRF700X driver for Matter.

>   * Reduced size of platform event queue.

Reviewer 2: ```suggestion
  * Reduced the size of Matter event queue.
```
